### ISSUE TITLE
Hints only mode

### ISSUE THREAD

**Issue by thriveth** [guotsuan/awesome-revelation#35]
When I use the only-this-tag option for revelation, it is confusing that it changes the layout to a different one containing the exact same clients. This goes against the principle of least surprise. I suggest to make it fall back to hints-only behavior when called for only the current tag.

**Comment by sammoth** [comment]
that wouldn't be much use if any clients were overlapping

**Comment by thriveth** [comment]
@sammoth so in floating mode, it makes sense to rearrange. But in tiling mode, it is confusing and unnecessary. Maybe it is possible to set the behavior depending on the layout? Or to rearrange if at least one floating client is present, otherwise not?

**Comment by sammoth** [comment]
makes sense, I guess it would only require no rearranging if the clients which are selected by the matcher functions, including minimised windows or those on other workspaces, are a subset of those which are visible and unobstructed on the screen currently. maybe a bit complex

**Comment by guotsuan** [comment]
Hi @thriveth,  I can see your point. The reason to rearrange the clients is to avoid the hint letters are too crowed for example. I am not happy with current "rearranging" too since it seems random.  I will work on that.  

Actually I myself use the hint only mode too because there is hints only model existing separately already. https://github.com/guotsuan/hints.  Although I think it is a good idea to merge them.
